common/src/main/java/com/wynntils/models/gearinfo/GearModel.java

```diff
@@ -0,0 +1,108 @@
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Model;
+import com.wynntils.models.gearinfo.itemguess.ItemGuessProfile;
+import com.wynntils.models.gearinfo.parsing.GearParseResult;
+import com.wynntils.models.gearinfo.parsing.GearParser;
+import com.wynntils.models.gearinfo.type.GearInfo;
+import com.wynntils.models.gearinfo.type.GearInstance;
+import com.wynntils.models.items.items.game.CraftedGearItem;
+import com.wynntils.models.items.items.game.GearItem;
+import com.wynntils.models.stats.StatModel;
+import com.wynntils.utils.type.CappedValue;
+import java.util.List;
+import java.util.regex.Matcher;
+import net.minecraft.world.item.ItemStack;
+
+/**
+ * Gear and stats are complex, have lots of corner cases and suffer from a general
+ * lack of comprehensible, exhaustive, correct and authoritive documentation. :-(
+ *
+ * Here is a collection of generally helpful links:
+ *
+ * 2016 Guide: https://forums.wynncraft.com/threads/how-identifications-are-calculated.128923/
+ * 2019 Guide: https://forums.wynncraft.com/threads/stats-and-identifications-guide.246308/
+ * The Damage Bible: https://docs.google.com/document/d/1BXdLrMWj-BakPcAWnuqvSFbwiz7oGTOMcEEdC5vCWs4
+ * WynnBuilder "Wynnfo": https://hppeng-wynn.github.io/wynnfo/, especially
+ * Damage Calculations: https://hppeng-wynn.github.io/wynnfo/pdfs/Damage_calculation.pdf
+ *
+ * A note on percent vs raw numbers and how they combine, from HeyZeer0:
+ * base = base + (base * percentage1) + (base * percentage2) + rawValue
+ */
+public final class GearModel extends Model {
+    private final GearInfoRegistry gearInfoRegistry = new GearInfoRegistry();
+
+    private final GearChatEncoding gearChatEncoding = new GearChatEncoding();
+
+    public GearModel(StatModel statModel) {
+        super(List.of(statModel));
+
+        ItemGuessProfile.init();
+    }
+
+    public void reloadData() {
+        gearInfoRegistry.reloadData();
+    }
+
+    public GearInstance parseInstance(GearInfo gearInfo, ItemStack itemStack) {
+        GearParseResult result = GearParser.parseItemStack(itemStack, gearInfo);
+        if (result.tier() != gearInfo.tier()) {
+            WynntilsMod.warn("Tier for " + gearInfo.name() + " is reported as " + result.tier());
+        }
+
+        return GearInstance.create(gearInfo, result.identifications(), result.powders(), result.tierCount());
+    }
+
+    public GearInstance parseInstance(GearInfo gearInfo, JsonObject itemData) {
+        GearParseResult result = GearParser.parseInternalRolls(gearInfo, itemData);
+
+        return GearInstance.create(gearInfo, result.identifications(), result.powders(), result.tierCount());
+    }
+
+    public CraftedGearItem parseCraftedGearItem(ItemStack itemStack) {
+        GearParseResult result = GearParser.parseItemStack(itemStack, null);
+        CappedValue durability = new CappedValue(result.tierCount(), result.durabilityMax());
+        // FIXME: Damages and requirements are not yet parsed
```

Comment:
Is this upcoming after this whole rewrite?

Reply:
I guess so. It's not *that* hard do to. But it kind of depends on what we want to do with the data! I don't even know how much we should care about crafted gear, or what do do with it. I had some idea of a "builder", and for that to know your exact stats, it needs to fully understand crafted gear. Otherwise, I frankly don't know. We don't actually use the identified stats we're parsing out of crafted gear right now either, it was more a matter of "well, it's simple to do, so why not do it?"